src/lua/datetime.lua

```diff
@@ -864,6 +879,86 @@ local function datetime_interval_add(lhs, rhs)
     end
 end
 
+--[[
+    Parse partial ISO-8601 date string
+
+    Accepted formats are:
+
+    Basic      Extended
+    20121224   2012-12-24   Calendar date   (ISO 8601)
+    2012359    2012-359     Ordinal date    (ISO 8601)
+    2012W521   2012-W52-1   Week date       (ISO 8601)
+    2012Q485   2012-Q4-85   Quarter date
+
+    Returns pair of constructed datetime object, and length of string
+    which has been accepted by parser.
+]]
+local function datetime_parse_date(str)
+    check_str(str, 'datetime.parse_date()')
+    local dt = ffi.new('dt_t[1]')
```

Comment:
I think it's better to use ffi stash for temporary objects (https://github.com/tarantool/tarantool/commit/a5549a44db835a59ad6ab82d206172f9d8047a2d)

Reply:
Done

```diff
--- a/src/lua/datetime.lua
+++ b/src/lua/datetime.lua
@@ -140,6 +140,10 @@ local date_strf_stash =
 local date_strf_stash_take = date_strf_stash.take
 local date_strf_stash_put = date_strf_stash.put
 
+local date_dt_stash = buffer.ffi_stash_new('dt_t[1]')
+local date_dt_stash_take = date_dt_stash.take
+local date_dt_stash_put = date_dt_stash.put
+
 local datetime_t = ffi.typeof('struct datetime')
 
 --[[
@@ -895,12 +899,15 @@ end
 ]]
 local function datetime_parse_date(str)
     check_str(str, 'datetime.parse_date()')
-    local dt = ffi.new('dt_t[1]')
+    local dt = date_dt_stash_take()
     local len = tonumber(builtin.tnt_dt_parse_iso_date(str, #str, dt))
     if len == 0 then
+        date_dt_stash_put(dt)
         error(('invalid date format %s'):format(str), 2)
     end
-    return datetime_new_dt(dt[0]), tonumber(len)
+    local d = datetime_new_dt(dt[0])
+    date_dt_stash_put(dt)
+    return d, tonumber(len)
 end
 
 --[[
```